src/auth_v3.js

```diff
@@ -149,12 +149,25 @@ export function getClient (cozy, clientParams) {
   if (!cli.isRegistered()) {
     return Promise.reject(new Error('Client not registered'))
   }
-  return cozyFetchJSON(cozy, 'GET', `/auth/register/${cli.clientID}`, null, {
-    manualAuthCredentials: {
-      token: cli
-    }
-  })
-    .then((data) => createClient(data, cli))
+  return cozyFetchJSON(cozy, 'GET', `/auth/register/${cli.clientID}`, null,
+    {
+      manualAuthCredentials: {
+        token: cli
+      }
+    })
+    .catch((err) => {
+      // If we fall into an error while fetching the client (because of a
+      // bad connectivity for instance), we do not bail the whole process
+      // since the client should be able to continue with the persisted
+      // client and token.
+      //
+      // If it is an explicit Unauthorized error though, we bail, clear th
+      // cache and retry.
+      if (FetchError.isUnauthorized(err) || FetchError.isNotFound(err)) {
```

Comment:
If `getClient` returns a 404, it means that the client does not exist. Either it has been revoked, or the client has never existed on this instance (if you take the client_id of one instance and call `getClient` on another instance for example). The effect is similar: you can't use the client and you should treat that as a revoked client.